src/redux/rootSaga.js

```diff
@@ -11,27 +11,92 @@ import {
   getDataFailure,
 } from './reducers/data';
 
-const pinUrl = `https://${process.env.DB_URL}/pins`;
+/* /////////// INDIVIDUAL API CALLS /////////// */
+
+const BASE_URL = process.env.DB_URL;
+
+async function getPins(filters) {
```

Comment:
since we are using Redux saga to handle our async calls we should use their generator function pattern rather than async/await. its pretty similar syntax so hopefully this breakdown helps:

```
async function getPins() => function*  getPins()
await => yield
await myFunc(arg1, arg2) => yield call(myFunc, arg1, arg2)
return => yield put($REDUX_ACTION)
```

Reply:
Hi Josh, I pushed a new commit converting the asyncs to generators.

One thing though, the four api-calling functions (getPins, getCounts, getFrequency, getTimeToClose) don't yield actions at the end because they're all being called under the umbrella of the main GET_DATA_REQUEST action that's dispatched when you hit the submit button. 

Basically we're making four api calls at the same time (and in parallel) to get all the data for the map and the visualizations. I discussed that with Adam on Tuesday, and we thought this would result in a better user experience than splitting up the calls, since the user won't see loading indicators when switching between the map and visualization tabs. Plus, doing separate calls for the map and visualizations could result in the filters getting out of sync with what's actually being shown.

As for the error and loading states, they're being handled by the `isLoading` and `error` fields in the data reducer. Right now those states are being shared between the map and all the charts, although at some point it might make sense to separate them out.

Reply:
Thanks for clarifying, I wasn't 100% sure what was going on. This seems like a solid solution for now esp for UI loading states, but i wonder if we should eventually split these calls into their own sagas to handle each API outcomes individually. Not sure what would happen if one call fails and the rest succeed. Probably worth a talk on tuesday about smarter API management, but lets go with this for now.